tests/unit_tests/arelle/test_qname.py

```diff
@@ -0,0 +1,187 @@
+import pytest
+from arelle.ModelValue import QName
+from itertools import product
+
+prefixes = ['pre', 1, '', None]
```

Comment:
Since the typing doesn't prevent us from calling it with bad data, I think its reasonable that we test that something reasonable happens when bad data is passed, however I am up for discussion.

Reply:
Agreed, but that changes behavior which I am specifically trying to avoid for anything other than already existing error cases